railties/lib/rails/generators/rails/benchmark/benchmark_generator.rb

```diff
@@ -0,0 +1,37 @@
+# frozen_string_literal: true
+
+require "rails/generators/named_base"
+
+module Rails
+  module Generators
+    class BenchmarkGenerator < NamedBase
+      argument :reports, type: :array, default: []
+      class_option :method, type: :string, enum: %w(ips bm bmbm), default: "ips"
+
+      def generate_layout
+        if benchmark_method == "ips"
+          add_ips_to_gemfile unless ips_installed?
+        end
+
+        template("benchmark.rb.tt", "benchmarks/#{file_name}.rb")
+      end
+
+    private
+      def benchmark_method
+        options["method"]
+      end
+
+      def add_ips_to_gemfile
+        gem_group(:development, :test) do
+          gem("benchmark-ips")
+        end
+      end
+
+      def ips_installed?
+        in_root do
+          return File.read("Gemfile").include?("gem 'benchmark-ips'")
```

Comment:
Do we need the `return` here?

Reply:
The `return` is needed as we're in the block scope of `in_root`. If I remove it the tests fail.

Reply:
[`in_root`](https://github.com/erikhuda/thor/blob/v1.0.0/lib/thor/actions.rb#L196) doesn't appear to pay attention to the return of the yield. This assumes the quoting of the gemfile. Can we do a regex like https://github.com/rails/rails/blob/d8b86e4b746962aa3b4c16cc447d9b48160c39f6/railties/lib/rails/generators/rails/db/system/change/change_generator.rb#L53 instead?

Reply:
@gmcgibbon I addressed this concern in a new commit.

👍 Good catch on the assumption I had with the gemfile's quoting preference.